File: go/internal/web/privacy.go
package web

import (
	"html/template"
	"net/http"
)

// contactEmailB64 is "closegamer@mail.ru" base64-encoded, matching the
// original Python component — never appears in plain text in the HTML
// source, only decoded client-side on click (see static/js/privacy.js).
const contactEmailB64 = "Y2xvc2VnYW1lckBtYWlsLnJ1"

type PrivacyPageData struct {
	PageData
	Text          template.HTML
	ContactEmailB64 string
}

// PrivacyHandler renders /privacy. Text is trusted, static HTML (headings,
// <br>, a link to Yandex's own privacy policy) ported verbatim from
// app/lang/*.py's privacy.text. The original's iframe-sandboxed email
// reveal widget (Streamlit components.html) becomes plain inline
// markup+JS here — no sandbox needed since the Go page isn't embedded.
func PrivacyHandler() http.HandlerFunc {
	return func(w http.ResponseWriter, r *http.Request) {
		lang := langFromRequest(r)
		data := &PrivacyPageData{
			PageData: NewPageData(
				lang,
				i18nOr(lang, "seo.title.privacy", "Privacy Policy - sConvert"),
				i18nOr(lang, "seo.description.privacy", ""),
				"https://sconvert.ru/privacy",
			),
			Text:            template.HTML(i18nOr(lang, "privacy.text", "")),
			ContactEmailB64: contactEmailB64,
		}
		Render(w, "page:privacy", data)
	}
}

File: go/cmd/server/static/js/privacy.js
/**
 * Contact email reveal widget. The address is base64-encoded in the HTML
 * source (data-email-b64) so it never appears in plain text to scrapers;
 * decoded client-side only when the user clicks "Contact". Ported from
 * app/components/privacy.py's Streamlit iframe widget — no iframe needed
 * here since the Go page isn't sandboxed.
 */
(function () {
  "use strict";

  document.addEventListener("DOMContentLoaded", function () {
    var wrap = document.getElementById("sc-privacy-email-wrap");
    var btn = document.getElementById("sc-privacy-reveal-btn");
    var text = document.getElementById("sc-privacy-email-text");
    if (!wrap || !btn || !text) return;

    var copiedLabel = wrap.getAttribute("data-copied-label") || "Copied";

    btn.addEventListener("click", function () {
      var email;
      try {
        email = atob(wrap.getAttribute("data-email-b64") || "");
      } catch (e) {
        return;
      }
      text.textContent = email;
      text.hidden = false;
      text.title = email;
      btn.hidden = true;

      text.addEventListener("click", function () {
        navigator.clipboard.writeText(email).then(function () {
          var original = text.textContent;
          text.textContent = copiedLabel;
          setTimeout(function () { text.textContent = original; }, 1200);
        }).catch(function () {});
      });
    });
  });
})();
